src/client/ua_client_connect.c

```diff
@@ -1636,6 +1829,7 @@ UA_Client_disconnectAsync(UA_Client *client) {
 
 UA_StatusCode
 UA_Client_disconnectSecureChannel(UA_Client *client) {
+    closeListeningReverseConnectSocket(client);
```

Comment:
You don't need `closeListeningReverseConnectSocket`.
The normal `closeSecureChannel` calls UA_SecureChannel_shutdown which call connectionManager->closeConnection.
That works just the same for the listening socket.

Reply:
Done